api/v1alpha1/serverclaim_types.go

```diff
@@ -44,11 +44,15 @@ const (
 type ServerClaimStatus struct {
 	// Phase represents the current phase of the server claim.
 	Phase Phase `json:"phase,omitempty"`
+
+	// ClaimedServerRef is a reference to a specific server which is claimed.
+	// This field is optional and can be omitted if the server is to be selected using ServerSelector.
+	ClaimedServerRef *v1.LocalObjectReference `json:"claimedServerRef,omitempty"`
```

Comment:
I guess it would make sense to have here a `name` + `UID` combination here as this is the technical representation of the resource we are claiming. You could avoid situation were a `Server` with the same name but a different `UID` is claimed.

Reply:
We could use here something like this
```
// LocalUIDReference is a reference to another entity including its UID
// +structType=atomic
type LocalUIDReference struct {
	// Name is the name of the referenced entity.
	Name string `json:"name"`
	// UID is the UID of the referenced entity.
	UID types.UID `json:"uid"`
}
```

/ref https://github.com/ironcore-dev/ironcore/blob/main/api/common/v1alpha1/common_types.go#L69

I would declare this type in the `common` file.